Serialisation API addition and Unique ID for Websocket JSON logging

ustulation opened the pull request:
<!-- Reviewable:start -->

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/maidsafe/maidsafe_utilities/51)

<!-- Reviewable:end -->

ustulation committed:
Unique ID for Websocket JSON logging

maidsafe-highfive commented:
r? @canndrew

(maidsafe_highfive has picked a reviewer for you, use r? to override)

ustulation commented:
r? @Viv-Rajkumar

ustulation committed:
Serialisation and Deserialisation with size limits - this is in accordance with the discussion on slack:

Spandan Sharma [10:26 AM]
So the error which i was talking about in morning catch-up { panic in test cases where data of bigger sizes is given to S.E } is basically due to serialisation size limit put here:
https://github.com/maidsafe/maidsafe_utilities/blob/master/src/serialisation.rs#L23
Does anybody know why it's there ?

[10:26]
(Anything >= 2MB will fail with that limit)

[10:27]
^^^ @group ^^^

Viv Rajkumar [10:28 AM]
ah that might have been set to avoid messages larger than that being sent in the network

[10:28]
but its ofc a problem for this

Spandan Sharma [10:28 AM]
Ohk

[10:29]
Could i remove it or should i comment those test cases for resolution later

[10:29]
(test cases in client i am writing)

Adam Cigánek [10:30 AM]
I think it's a security measure. Without it, it would be possible to send a crafted message that would crash the app. I wouldn't remove it, but it should be fine to increase it. Ask @andrew.cann for more details.

Spandan Sharma [10:36 AM]
Ok - The problem is of-course with the mocks - i need to read and write to the file on the disk to simulate vault storage - that file could get much bigger than 2 MBs (because data is persistent and frontend demo apps and retrieve it later etc.)

[10:37]
So i can go ahead and use serialisation without size limit for the mock code - should be fine i guess.

Adam Cigánek [10:38 AM]
Yes, if it's mock only, I think it should be fine.

Spandan Sharma [10:38 AM]
thanks !

Spandan Sharma [11:15 AM]
@adam: @andrew.cann sorry just one more point (as i copy paste the serialisation code into mock) , shouldn't these size limits be somewhere in the networking layer / transport layer ?

[11:16]
it looks out of place for that restriction to be in a utility library which can be used for different purposes

[11:17]
Exactly the kind of problem i am facing - even though we have the code i am unable to use it and need to replicate it for reasons that does not concern everyone (every lib)

Adam Cigánek [11:22 AM]
I agree. I'd suggest adding new functions, say `serialise_with_limit` and `deserialise_with_limit`, or something like that.

[11:22]
And keep the existing one using the default limit, to preserve the API

Spandan Sharma [11:23 AM]
cool - you make my day :smile: - (Although i suspect this will need revisit after things settle down - OKish for now)

Viv-Rajkumar commented:
Reviewed 3 of 3 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/maidsafe/maidsafe_utilities/51)*
<!-- Sent from Reviewable.io -->
